src/strings/ops.c

```diff
@@ -206,38 +205,130 @@ static void turn_32bit_into_8bit_unchecked(MVMThreadContext *tc, MVMString *str)
 
     MVM_free(old_buf);
 }
+/* Checks if the next num_graphs graphemes in the iterator can fit into 8 bits.
+ * This was written to take advantage of SIMD vectorization, so we use a multiple
+ * bitwise operations to check, and biwise OR it with val. Care must be taken
+ * to not use any variables altered by the loop outside of the loop and to not
+ * have any branching or funcion calls. `i` is not used outside the loop
+ * `val` is allowed as biwise OR works with the vectorization well.
+ * NOTE: Final state of the grapheme iterator passed into i is undefined. It should
+ * not be reused. */
+static int string_can_be_8bit(MVMThreadContext *tc, MVMGraphemeIter *gi, MVMStringIndex num_graphs) {
+    int val = 0;
+    MVMStringIndex pos = 0;
+    while (1) {
+        size_t strand_len = gi->end - gi->pos;
+        if (gi->blob_type == MVM_STRING_GRAPHEME_32) {
+            size_t i = 0;
+            size_t togo     = strand_len > num_graphs - pos ? num_graphs - pos : strand_len;
+            size_t togofull = togo + gi->pos;
+            for (i = gi->pos; i  < togofull; i++) {
+                val |= ((gi->active_blob.blob_32[i] & 0xffffff80) + 0x80) & (0xffffff80-1);
+            }
+            if (val) return 0;
+            pos += togo;
+        }
+        else {
```

Comment:
Both statements in this else branch are also in the if branch. Couldn't both be brought out of the if/else entirely, with `size_t togo = strand_len > num_graphs - pos ? num_graphs - pos : strand_len;` going before the if and `pos += togo;` going after the if (and the else going away). I.e.,
```
size_t togo = strand_len > num_graphs - pos ? num_graphs - pos : strand_len;
if (gi->blob_type == MVM_STRING_GRAPHEME_32) {
    size_t i = 0;
    size_t togofull = togo + gi->pos;
    for (i = gi->pos; i  < togofull; i++) {
        val |= ((gi->active_blob.blob_32[i] & 0xffffff80) + 0x80) & (0xffffff80-1);
    }
    if (val) return 0;
}
pos += togo;
```